excise/utils.py

```diff
@@ -423,3 +394,54 @@ def get_order_tax_data(
                 )
         raise TaxError(customer_msg)
     return response
+
+
+def _retrieve_meta_data_from_cache(token):
+    cached_data = cache.get(token)
+    return cached_data
+
+def metadata_requires_update (
+    metadata: str,
+    token_in_cache: str,
+    force_refresh: bool = False,
+):
+    """Check if Checkout metadata needs to be reset.
+
+    The itemized taxes from ATE are stored in a cache. If an object doesn't exist in cache
+    or something has changed, taxes need to be refetched.
+    """
+    if force_refresh:
+        return True
+
+    cached_metadata = _retrieve_meta_data_from_cache(token_in_cache)
+
+    if not cached_metadata:
+        return True
+
+    if cached_metadata != metadata:
+        return True
+
+    return False
+
+def process_checkout_metadata(
+    metadata: str,
+    checkout_token: str,
+    force_refresh: bool = False,
+    cache_time: int = CACHE_TIME
+):
+    """Check for Checkout metadata in cache.
+
+    Do nothing if metadata are the same. Set new metadata in other cases.
+    """
+    data_cache_key = "checkout_metadata_" + str(checkout_token)
```

Comment:
☝️ up. 
The cache is shared between all plugins and Saleor instance. For someone who doesn't know the current implementation ATEPlugin, it will be hard to debug (or anything) with a key name like this. The key name should contain at least something which we can connect with ATE plugin.

Reply:
namespaced this key to our plugin